azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -15,34 +15,405 @@
+  private final String clusterName;
+  private final String flowContainerName;
+  private final String cpuLimit;
+  private final String cpuRequest;
+  private final String memoryLimit;
+  private final String memoryRequest;
+
+
+  private static final Logger logger = LoggerFactory
+      .getLogger(KubernetesContainerizedImpl.class);
 
   @Inject
-  KubernetesContainerizedImpl(final Props azkProps, final ExecutorLoader executorLoader) throws ExecutorManagerException {
+  public KubernetesContainerizedImpl(final Props azkProps, final ExecutorLoader executorLoader)
+      throws ExecutorManagerException {
     this.azkProps = azkProps;
     this.executorLoader = executorLoader;
+    this.namespace = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_NAMESPACE);
+    this.flowContainerName =
+        azkProps.getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_NAME
+            , DEFAULT_FLOW_CONTAINER_NAME_PREFIX);
+    this.podPrefix =
+        azkProps.getString(ContainerizedExecutionManagerProperties.KUBERNETES_POD_NAME_PREFIX,
+            DEFAULT_POD_NAME_PREFIX);
+    this.servicePrefix = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_SERVICE_NAME_PREFIX,
+            DEFAULT_SERVICE_NAME_PREFIX);
+    this.clusterName = azkProps.getString(ConfigurationKeys.AZKABAN_CLUSTER_NAME,
+        DEFAULT_CLUSTER_NAME);
+    this.cpuLimit = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_CPU_LIMIT,
+            CPU_LIMIT);
+    this.cpuRequest = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_CPU_REQUEST,
+            DEFAULT_CPU_REQUEST);
+    this.memoryLimit = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_MEMORY_LIMIT,
+            MEMORY_LIMIT);
+    this.memoryRequest = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_MEMORY_REQUEST,
+            DEFAULT_MEMORY_REQUEST);
+
+    try {
+      // Path to the configuration file for Kubernetes which contains information about
+      // Kubernetes API Server and identity for authentication
+      String kubeConfigPath = azkProps
+          .getString(ContainerizedExecutionManagerProperties.KUBERNETES_KUBE_CONFIG_PATH);
+      logger.info("Kube config path is : {}", kubeConfigPath);
+      this.client =
+          ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(
+              Files.newBufferedReader(Paths.get(kubeConfigPath), Charset.defaultCharset())))
+              .build();
+      this.coreV1Api = new CoreV1Api(client);
+    } catch (IOException exception) {
+      logger.error("Unable to read kube config file: {}", exception.getMessage());
+      throw new ExecutorManagerException(exception);
+    }
   }
 
+  /**
+   * This method is used to create container during dispatch of execution. It will create pod for a
```

Comment:
I guess we are using pod and container interchangeably here which may not be true in the future.
At this stage, it is pod so should we stick to calling it pod only?

Reply:
Container is a common word for any containerized infrastructure. Pod is very specific to Kubernetes. That is why createContainer is kept in interface. I have added comment to explain what createContainer is for Kubernetes.